src/record_ts.cpp

```diff
@@ -0,0 +1,1446 @@
+                                 info.slot, info.pointer_access, size);
+                    } else {
+                        jitc_log(LogLevel::Info, " <- param s%u is_pointer=%u",
+                                 info.slot, info.pointer_access);
+                    }
+
+                    if (info.type == ParamType::Output) {
+                        rv.alloc(backend, launch_size, info.vtype);
+                    }
+                    jitc_log(LogLevel::Info, "    data=%p", rv.data);
+                    jitc_assert(
+                        rv.data != nullptr || dry_run,
+                        "replay(): Encountered nullptr in kernel parameters.");
+                    kernel_params.push_back(rv.data);
+                }
+
+                // Change kernel size in `kernel_params`
+                if (backend == JitBackend::CUDA) {
+                    uintptr_t size = 0;
+                    std::memcpy(&size, &launch_size, sizeof(uint32_t));
+                    kernel_params[0] = (void *) size;
+                }
+
+                if (!dry_run) {
+                    jitc_log(LogLevel::Debug,
+                             "replay(): launching kernel %u %016llx [%u]%s",
+                             n_kernels++,
+                             (unsigned long long) op.kernel.hash.high64,
+                             launch_size, op.uses_optix ? " uses optix" : "");
+
+                    std::vector<uint32_t> kernel_param_ids;
+                    std::vector<uint32_t> kernel_calls;
+                    Kernel kernel = op.kernel.kernel;
+                    if (op.uses_optix) {
+                        uses_optix    = true;
+                        ts->optix_sbt = op.sbt;
+                    }
+                    ts->launch(kernel, op.kernel.key, op.kernel.hash,
+                               launch_size, &kernel_params, &kernel_param_ids);
+                    if (op.uses_optix)
+                        uses_optix = false;
+                }
+
+            };
+            break;
+            case OpType::Barrier: {
+                ProfilerPhase profiler(pr_barrier);
+                if (!dry_run)
+                    ts->barrier();
+            };
+            break;
+            case OpType::MemsetAsync: {
+                ProfilerPhase profiler(pr_memset_async);
+
+                uint32_t dependency_index = op.dependency_range.first;
+
+                ParamInfo ptr_info = this->dependencies[dependency_index];
+
+                ReplayVariable &ptr_var = replay_variables[ptr_info.slot];
+                ptr_var.alloc(backend, op.size, op.input_size);
+
+                jitc_log(LogLevel::Debug,
+                         "replay(): MemsetAsync -> s%u [%zu], "
+                         "op.input_size=%zu",
+                         ptr_info.slot, op.size, op.input_size);
+
+                uint32_t size = ptr_var.size(op.input_size);
+
+                if (!dry_run)
```

Comment:
I guess the rest of the cases' code is still executed in dry_run mode so that you can see the logs?
Other than the log, it's a no-op right?

Reply:
The rest of the `replay_*` functions might still be performing size inference, for example calling `rv.alloc(backend, size)` will not allocate in dry run mode, but will change the `dsize` of the variable. Calling `rv.size(vartype)` in the next operation will then calculate the size from the memory size and type.

Should I add a comment?